### PR TITLE
ChooserWidget: Trigger manually change event when input.value is set

### LINKED ISSUE #10186: Support django-smart-selects for snippet models
### Is your proposal related to a problem?

I'd like to implement cascading/chained dropdown lists in the Wagtail admin. For example let's say there is a `Location` model with foreign keys to `Country` and `City`. Once a user selects a specific country, the city dropdown is updated to contain only the cities for this country. [django-smart-selects](https://github.com/jazzband/django-smart-selects) works great in the Django admin, and also in the Wagtail admin for regular Django models, but not if the linked model(s) (`Country` in the example) is registered as a Wagtail snippet. 

### Describe the solution you'd like

After digging into it, `django-smart-selects` [sets a handler on the change event](https://github.com/jazzband/django-smart-selects/blob/master/smart_selects/static/smart-selects/admin/js/chainedfk.js#L89). This event is fired for non-snippet models because the respective element is a `<select>`. However it is not fired for snippet models because the respective element is `<input type="hidden">` and [changes to hidden elements don't automatically fire the .change() event](https://stackoverflow.com/a/8965804). Based on the linked answer, the solution is to tell jQuery to trigger it explicitly (`.trigger('change')`).

### PR DESCRIPTION
**Pull request by gsakkis**
Fixes #10186

_Please check the following:_

-   [ ] Do the tests still pass?[^1]
-   [ ] Does the code comply with the style guide?
    -   [ ] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Review by lb- — COMMENTED** [review]
Should these events bubble? So that they emulate actual change events?

Or would a custom event name be suitable for what you need?

**Comment by gsakkis** [comment]
Yes they should emulate actual change events; as mentioned in #10186, django-smart-selects [sets a handler on the change event](https://github.com/jazzband/django-smart-selects/blob/master/smart_selects/static/smart-selects/admin/js/chainedfk.js#L89).

File: client/src/components/ChooserWidget/index.js [hunk 1]
```diff
@@ -85,11 +85,13 @@ export class Chooser {
 
   renderEmptyState() {
     this.input.setAttribute('value', '');
+    this.input.dispatchEvent(new Event("change"));
```
Comment (lb-) [inline client/src/components/ChooserWidget/index.js:88]:
```suggestion
    this.input.dispatchEvent(new Event("change", { bubbles: true }));
```

File: client/src/components/ChooserWidget/index.js [hunk 2]
```diff
@@ -85,11 +85,13 @@ export class Chooser {
...
     this.chooserElement.classList.add('blank');
   }
 
   renderState(newState) {
     this.input.setAttribute('value', newState.id);
+    this.input.dispatchEvent(new Event("change"));
```
Comment (lb-) [inline client/src/components/ChooserWidget/index.js:94]:
```suggestion
    this.input.dispatchEvent(new Event("change", { bubbles: true }));
```

**Review by lb- — CHANGES_REQUESTED** [review]
Haven't tested yet but as noted, this should probably bubble.

File: client/src/components/ChooserWidget/index.js (hunk 2 above)
Comment (gsakkis) [reply client/src/components/ChooserWidget/index.js]: Done

File: client/src/components/ChooserWidget/index.js (hunk 1 above)
Comment (gsakkis) [reply client/src/components/ChooserWidget/index.js]: Done

**Review by lb- — APPROVED** [review]
Looking good, I have tested before and after and can see that choosers now correctly fire a `'change'` event.

Really appreciate this change. I have run the formatter and will push up a rebase of this to merge in.

I will add you to the contributors list as 'George Sakkis' but let me know if you want a different name listed.

### Some validation screenshots

Screenshot 2023-03-11 at 2 52 29 pm
Screenshot 2023-03-11 at 2 51 53 pm

**Pull request merged**